### ISSUE TITLE
ProxyValidator does not parse NO_PROXY correctly when looking for hosts

### ISSUE THREAD

**Issue by kmod-midori** [flutter/flutter#35598]
## Use case
`flutter doctor` was unable to detect the problem when I wrote `localhost;127.0.0.1` in `NO_PROXY`, note the semicolon instead of comma.

## Proposal
As there is no standard way of delimiting values in environment variables (for example, on Windows, `;` is usually used), clear instructions should be given about setting `NO_PROXY`.

**Comment by kangwang1988** [comment]
According to the title, I think that you want the NO_PROXY detection logic should be able to the error if it's set improperly like "localhost;127.0.0.1", is that right?
Now the check logic is pretty simple:
```dart
        for (String host in const <String>['127.0.0.1', 'localhost']) {
          final ValidationMessage msg = _noProxy.contains(host)
              ? ValidationMessage('NO_PROXY contains $host')
              : ValidationMessage.hint('NO_PROXY does not contain $host');

          messages.add(msg);
        }
```

**Comment by chengyuhui** [comment]
Yes, that could have saved me an hour of searching and head scratching.

**Comment by kangwang1988** [comment]
cc @zoechi 
Maybe update the document about NO_PROXY setting.

**Comment by Hixie** [comment]
The tool should do the splitting on commas and compare that, instead of looking for substrings.
